src/libraries/System.Runtime.InteropServices/src/System.Runtime.InteropServices.csproj

```diff
@@ -48,7 +50,23 @@
     <Compile Include="System\Security\SecureStringMarshal.cs" />
     <Compile Include="$(CommonPath)System\Obsoletions.cs" Link="Common\System\Obsoletions.cs" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetsUnix)' == 'true'">
+    <Compile Include="System\Runtime\InteropServices\PosixSignalRegistration.Unix.cs" />
+    <Compile Include="$(CommonPath)Interop\Unix\System.Native\Interop.PosixSignal.cs"
+             Link="Common\Interop\Unix\Interop.PosixSignal.cs" />
+    <Compile Include="$(CommonPath)Interop\Unix\System.Native\Interop.InitializeTerminalAndSignalHandling.cs"
+             Link="Common\Interop\Unix\Interop.InitializeTerminalAndSignalHandling.cs" />
+    <Compile Include="$(CommonPath)Interop\Unix\Interop.Libraries.cs"
+             Link="Common\Interop\Unix\Interop.Libraries.cs" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetsUnix)' != 'true'">
+    <Compile Include="System\Runtime\InteropServices\PosixSignalRegistration.cs" />
+  </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="$(CoreLibProject)" />
   </ItemGroup>
+  <ItemGroup  Condition="'$(TargetsUnix)' == 'true'">
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Runtime\src\System.Runtime.csproj" />
+    <Reference Include="Microsoft.Win32.Primitives" />
```

Comment:
Added these by _trail and error_ to be able to throw `Win32Exception`. I'm not sure this is the proper way.

Reply:
I think these cause weird errors further down the build:
```
/__w/1/s/.packages/microsoft.dotnet.sharedframework.sdk/6.0.0-beta.21311.3/targets/sharedfx.targets(467,5): error : Cycle detected for /__w/1/s/artifacts/bin/pkg/net6.0/runtime/Linux-Debug-arm/System.Runtime.InteropServices.dll. [/__w/1/s/src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Runtime.sfxproj]
##[error].packages/microsoft.dotnet.sharedframework.sdk/6.0.0-beta.21311.3/targets/sharedfx.targets(467,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) Cycle detected for /__w/1/s/artifacts/bin/pkg/net6.0/runtime/Linux-Debug-arm/System.Runtime.InteropServices.dll.
/__w/1/s/.packages/microsoft.dotnet.sharedframework.sdk/6.0.0-beta.21311.3/targets/sharedfx.targets(467,5): error : System.Runtime.InteropServices > Microsoft.Win32.Primitives > System.Runtime.InteropServices [/__w/1/s/src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Runtime.sfxproj]
##[error].packages/microsoft.dotnet.sharedframework.sdk/6.0.0-beta.21311.3/targets/sharedfx.targets(467,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) System.Runtime.InteropServices > Microsoft.Win32.Primitives > System.Runtime.InteropServices
```

Reply:
See https://github.com/dotnet/runtime/pull/54136#discussion_r651220521